### PR TITLE
insert dynamic and static with tests

### PR DESCRIPTION
**Pull request by jtlap**
I also added a may_alias_t, for orthogonality.

In sse2 only epi16 has an intrisic, perhaps it would be interesting to test the speed of 2 calls to insert a 32 bit data (4 to insert 64 bit) against the reinterpret cast version

### CODE REVIEW

File: include/boost/simd/arch/common/scalar/function/insert.hpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    {
+      a0 = a2;
+    }
+  };
+
+  BOOST_DISPATCH_OVERLOAD ( insert_
+                          , (typename A0, typename A1, typename A2)
+                          , bd::cpu_
+                          , bd::unspecified_<A0>
+                          , bd::constant_< bd::integer_<A1>>
```
Comment (jfalcou) [inline include/boost/simd/arch/common/scalar/function/insert.hpp]: in act as constant_ is a chidlren o scalar<integral> this specialization is maybe useless. Could you test ?

File: include/boost/simd/arch/x86/sse2/simd/function/insert.hpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                          )
+  {
+
+    BOOST_FORCEINLINE void operator() (  A0 & a0, A1, A2 const & a2) const BOOST_NOEXCEPT
+    {
+      a0 = _mm_insert_epi16(a0, a2, A1::value);
+    }
+
+  };
+
```
Comment (jfalcou) [inline include/boost/simd/arch/x86/sse2/simd/function/insert.hpp:39]: again, wh i   t    e       s   p       a       ce  s ;)

File: include/boost/simd/arch/x86/sse4_1/simd/function/insert.hpp [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                          , bs::sse4_1_
+                          , bs::pack_<bd::ints64_<A0>,bs::sse_>
+                          , bd::constant_< bd::integer_<A1>>
+                          , bd::scalar_< bd::unspecified_<A2>>
+                          )
+  {
+
+    BOOST_FORCEINLINE void operator() (  A0 & a0, A1, A2 const & a2) const BOOST_NOEXCEPT
+    {
+      a0 = _mm_insert_epi64(a0, a2, A1::value);
```
Comment (jfalcou) [inline include/boost/simd/arch/x86/sse4_1/simd/function/insert.hpp]: this maybe unreliable in debug mode. I would do : 

```
static const A1::type value = A1::value;
a0 = _mm_insert_epi64(a0, a2, value);
```

looks stupid but it fixed issues on ICC in old NT2

File: include/boost/simd/arch/common/scalar/function/insert.hpp (hunk 1 above)
Comment (jtlap) [inline include/boost/simd/arch/common/scalar/function/insert.hpp]: that looks ok to do

File: include/boost/simd/arch/x86/sse4_1/simd/function/insert.hpp (hunk 3 above)
Comment (jtlap) [inline include/boost/simd/arch/x86/sse4_1/simd/function/insert.hpp]: But here it adds an issue ! (even with typename)

/home/jt/boost_simd_dev/a_boost_simd/include/boost/simd/arch/x86/sse4_1/simd/function/insert.hpp:36:38: error: no viable conversion from 'const unsigned long' to 'const typename integral_constant<unsigned long, 15>::type' (aka 'const integral_constant<unsigned long, 15UL>')
      static const typename A1::type value  = A1::value;
                                     ^        ~~~~~~~~~
/home/jt/boost_simd_dev/build/clang++-3.6/_install/include/boost/dispatch/function/functor.hpp:65:43: note: in instantiation of member function 'boost::simd::ext::impl_insert_<boost::simd::sse4_1_, boost::simd::pack_<boost::dispatch::ints_<boost::simd::pack<signed char, 16, boost::simd::sse_>, 8>, boost::simd::sse_>, boost::dispatch::constant_<boost::dispatch::integer_<std::integral_constant<unsigned long, 15> > >, boost::dispatch::scalar_<boost::dispatch::unspecified_<signed char> > >::operator()' requested here
      return BOOST_DISPATCH_IMPL_TAG_CALL(Tag,Site,Args,args);

Comment (jfalcou) [inline include/boost/simd/arch/x86/sse4_1/simd/function/insert.hpp]: it's ::value_type sorry

**Pull request merged**